scripts/safari.js

```diff
@@ -6997,6 +6944,128 @@ function Safari() {
+            } else {
+                changedTeams[name] = "you have been moved to the " + toColor(otherTeam, chosenColor) + " side because the " + chosenTeam + " side was full!";
+            }
+        }
+        
+        var i, len = this.team1.length, player, member;
+        for (e = 0; e < len; e++) {
+            name = this.team1[e];
+            player = getAvatarOff(name);
+            this.playerTeams[name] = [];
+            for (i = 0; i < 6; i++) {
+                member = { owner: name, id: player.party[i], score: 0 };
+                this.party1.push(member);
+                this.playerTeams[name].push(member);
+            }
+        }
+        len = this.team2.length;
+        for (e = 0; e < len; e++) {
+            name = this.team2[e];
+            player = getAvatarOff(name);
+            this.playerTeams[name] = [];
+            for (i = 0; i < 6; i++) {
+                member = { owner: name, id: player.party[i], score: 0 };
+                this.party2.push(member);
+                this.playerTeams[name].push(member);
+            }
+        }
+        
+        var createFillerPlayer = function() {
+            var name = "[NPC] " + generateName();
+            var party = generateTeam();
+            return { name: name, party: party };
+        };
+        
+        var playerCount = this.signups.length;
+        if (playerCount < 4) {
+            this.hasReward = false;
+        }
+        
+        while (playerCount < 6 || playerCount % 2 === 1) {
+            var useSide1 = this.team1.length < this.team2.length;
+            var team = useSide1 ? this.team1 : this.team2;
+            var party = useSide1 ? this.party1 : this.party2;
+            
+            var npc = createFillerPlayer();
+            
+            if (!team.contains(npc.name) && !this.npcs.contains(npc.name)) {
+                team.push(npc.name);
+                this.npcs.push(npc.name);
+                this.playerTeams[npc.name] = [];
+                for (i = 0; i < 6; i++) {
+                    member = { owner: npc.name, id: npc.party[i], score: 0 };
+                    party.push(member);
+                    this.playerTeams[npc.name].push(member);
+                }
+                playerCount++;
+            }
+        }
+        
+        this.party1.shuffle();
+        this.party2.shuffle();
+        this.battleSize = this.party1.length;
+        
+        this.sendToViewers("");
+        safaribot.sendHtmlAll("The " + this.eventName + " is starting now! If you didn't join, you still can watch by typing " + link("/watch") + "!", safchan);
+        this.sendToViewers("The teams have been defined! ");
+        this.sendToViewers(toColor(this.team1Name, this.team1Color) + ": " + readable(this.team1.map(addFlashTag), "and"), true);
+        this.sendToViewers(toColor(this.team2Name, this.team2Color) + ": " + readable(this.team2.map(addFlashTag), "and"), true);
+        for (e in changedTeams) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.